File: backend/app/api/schemas/control_plane.py
"""Control plane API schema — v0.8.8."""
from typing import List, Optional
from pydantic import BaseModel


class PositionViewSchema(BaseModel):
    position_id: str
    event_key: str
    side: str
    status: str

    trigger_price: float
    entry_fill_price: float
    current_price: float
    exit_fill_price: float

    trigger_move_value: float
    fill_move_value: float
    current_move_value: float

    realized_pnl: float
    unrealized_pnl: float

    entry_reason: str
    exit_reason: str
    opened_at: str
    closed_at: Optional[str]


class ControlPlaneResponse(BaseModel):
    # Positions
    open_positions: List[PositionViewSchema]
    closed_positions: List[PositionViewSchema]

    # Session PnL
    session_realized_pnl: float
    session_unrealized_pnl: float
    session_total_pnl: float

    # Balance
    total_balance: float
    available_balance: float
    current_balance: float
    session_start_balance: float

    # Claim
    claim_status: str
    claim_available: bool
    claimed_amount: float
    settlement_completed_at: Optional[str]

    # Gate state
    release_ready: bool
    live_applied_testing_ready: bool
    live_mode_ui_blocked: bool
    blocked_reason_messages: List[str]

File: frontend/src/types/controlPlane.ts
/** Control plane state shape returned by GET /control-plane — v0.8.8 */

export interface PositionViewData {
  position_id: string
  event_key: string
  side: string
  status: string
  trigger_price: number
  entry_fill_price: number
  current_price: number
  exit_fill_price: number
  trigger_move_value: number
  fill_move_value: number
  current_move_value: number
  realized_pnl: number
  unrealized_pnl: number
  entry_reason: string
  exit_reason: string
  opened_at: string
  closed_at: string | null
}

export interface ControlPlaneState {
  open_positions: PositionViewData[]
  closed_positions: PositionViewData[]
  session_realized_pnl: number
  session_unrealized_pnl: number
  session_total_pnl: number
  total_balance: number
  available_balance: number
  current_balance: number
  session_start_balance: number
  claim_status: string
  claim_available: boolean
  claimed_amount: number
  settlement_completed_at: string | null
  release_ready: boolean
  live_applied_testing_ready: boolean
  live_mode_ui_blocked: boolean
  blocked_reason_messages: string[]
}
